include/rocksdb/file_system.h

```diff
@@ -825,6 +852,20 @@ class FSRandomAccessFile {
     return IOStatus::NotSupported("InvalidateCache not supported.");
   }
 
+  //  Asynchronous Read of the data requested in FSReadRequest. This is a
+  //  asynchronous call, i.e it will return after submitting the request.
+  //  When read request is completed callback function specified in cb will be
+  //  called with result populated in FSReadResponse. This API also populates
+  //  IOHandle. req contains the request offset and size, but the result and
+  //  status fields should be ignored. Instead, they should be passed in the
+  //  FSReadResponse structure to the callback.
+  virtual IOStatus ReadAsync(const IOOptions& /*opts*/, IODebugContext* /*dbg*/,
+                             FSReadRequest* /*req*/,
```

Comment:
1.  The order of arguments does not follow the convention of the rest of the APIs for this class;
2. I do not think you want this to have an FSReadRequest.  Instead you should probably have an offset and length.  The ReadRequest also contains the result data and the result Status.
3. The CB should take a const FSReadResponse&, not a pointer.